src/CMakeLists.txt

```diff
@@ -1,326 +1,129 @@
-    target_compile_definitions(raylib
-      PRIVATE $<BUILD_INTERFACE:BUILD_LIBTYPE_SHARED>
-      INTERFACE $<INSTALL_INTERFACE:USE_LIBTYPE_SHARED>
+# Get the sources together
+set(raylib_public_headers
+    raylib.h
+    rlgl.h
+    physac.h
+    raymath.h
+    raudio.h
     )
-  endif()
-
-  set(PKG_CONFIG_LIBS_EXTRA "")
-
-  set_property(TARGET raylib PROPERTY POSITION_INDEPENDENT_CODE ON)
-  set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_LIBDIR}")
-  set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-  set(CMAKE_MACOSX_RPATH ON)
 
-  target_link_libraries(raylib ${LIBS_PRIVATE})
-  if (${PLATFORM} MATCHES "Desktop")
-    target_link_libraries(raylib glfw)
-  endif()
-  set_target_properties(raylib PROPERTIES
-      PUBLIC_HEADER "raylib.h"
-      VERSION   ${PROJECT_VERSION}
-      SOVERSION ${API_VERSION}
-  )
-
-  if (WIN32)
-    install(
-      TARGETS raylib
-      ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-      LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-      RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
-      PUBLIC_HEADER DESTINATION "include"
-    )
-  else()
-    install(
-        TARGETS raylib
-        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-        PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+set(raylib_sources
+    core.c
+    models.c
+    shapes.c
+    text.c
+    textures.c
+    utils.c
     )
-  endif()
-  set_target_properties(raylib PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_BINARY_DIR}")
 
-  add_test("pkg-config" ${PROJECT_SOURCE_DIR}/../cmake/test-pkgconfig.sh)
-else(SHARED)
-  add_library(raylib ALIAS raylib_static)
-endif(SHARED)
+if (USE_AUDIO)
+    MESSAGE(STATUS "Audio Backend: miniaudio")
+    list(APPEND raylib_sources ${CMAKE_CURRENT_SOURCE_DIR}/raudio.c)
+else ()
+    MESSAGE(STATUS "Audio Backend: None (-DUSE_AUDIO=OFF)")
+endif ()
 
-if (NOT DEFINED PKG_CONFIG_LIBS_EXTRA)
-  set(PKG_CONFIG_LIBS_EXTRA "${PKG_CONFIG_LIBS_PRIVATE}")
-endif()
-
-join_paths(libdir_for_pc_file "\${exec_prefix}" "${CMAKE_INSTALL_LIBDIR}")
-join_paths(includedir_for_pc_file "\${prefix}" "${CMAKE_INSTALL_INCLUDEDIR}")
+include(LibraryConfigurations)
 
-configure_file(../raylib.pc.in raylib.pc @ONLY)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/raylib.pc DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
-configure_file(../cmake/raylib-config-version.cmake raylib-config-version.cmake @ONLY)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/raylib-config-version.cmake DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/raylib")
-install(FILES ${PROJECT_SOURCE_DIR}/../cmake/raylib-config.cmake DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/raylib")
+set(LIBS_PRIVATE ${LIBS_PRIVATE} ${OPENAL_LIBRARY})
 
-# populates raylib_{FOUND, INCLUDE_DIRS, LIBRARIES, LDFLAGS, DEFINITIONS}
-include(PopulateConfigVariablesLocally)
-populate_config_variables_locally(raylib)
+if (STATIC)
+    MESSAGE(STATUS "Building raylib static library")
+    
+    add_library(raylib STATIC ${raylib_sources})
+    add_library(raylib_static ALIAS raylib)
```

Comment:
raylib_static was a workaround, so when building on Windows the lib files didn't override each other. Does this still work as intended?

Reply:
@a3f I would say **No** but the reason being is that you would be interested in building only one or the other at a given time. It will be more clear and also best if we start relying on the **default provided** [BUILD_SHARED_LIBS](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html) variable instead. Using that we can only define one library target and then the user will be responsible on deciding what to build and how to separate the two. At least that is a more clean and clear approach for any modern cmake user -[ read this article that describes why](https://cgold.readthedocs.io/en/latest/tutorials/libraries/static-shared.html).

I also planned on refactoring to the right approach once these improvements get accepted.

Reply:
I guess we can drop it if it improves code clarity.